src/components/DownloadExperiment.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+
+import Button from './Button';
+import ProcessingGears from './ProcessingGears';
+import ModalManager from './Modal/ModalManager';
+import {
+  DatasetDownloadOptionsForm,
+  AggreationOptions,
+  TransformationOptions,
+  AdvancedOptions,
+  EmailAddressOptions,
+  SubmitDatasetOptionsForm,
+} from './DatasetDownloadOptionsForm';
+
+import { getDataSet } from '../api/dataSet';
+
+import { useLocalStorage } from '../common/hooks';
+import { getDomain } from '../common/helpers';
+
+const DownloadExperiment = ({ experiment, showProcessing = false }) => {
+  const [emailAddress] = useLocalStorage('email-address');
+
+  const {
+    accession_code: experimentAccessionCode,
+    num_downloadable_samples: downloadableSamplesCount,
+  } = experiment;
+
+  const data = {};
+  data[experimentAccessionCode] = experiment.samples.map(s => s.accession_code);
+
+  const emptyDataset = {
+    data,
+    aggregate_by: 'EXPERIMENT',
+    quantile_normalize: true,
+    scale_by: 'NONE',
+    email_address: undefined,
+    email_ccdl_ok: false,
+  };
+
+  const [dataset, setDataset] = React.useState(emptyDataset);
+
+  const [processingDataset, setDatasetIsProcessing] = useLocalStorage(
+    `dataset-${experiment.accession_code}`,
+    false
+  );
+
+  // check if dataset is being processed
+  // if not clean up localstorage
+  React.useEffect(() => {
+    const datasetId = dataset.id || processingDataset;
+    const refreshState = async () => {
+      const refreshedDataset = await getDataSet(datasetId);
+
+      if (refreshedDataset.is_processing) {
+        setDataset(refreshedDataset);
+        setDatasetIsProcessing(refreshedDataset.id);
+      } else {
+        setDataset(emptyDataset);
+        setDatasetIsProcessing(null);
+      }
+    };
+
+    const justStarted = dataset.id !== processingDataset;
```

Comment:
in what scenario is this `false`?

Reply:
This is false when the id's are the same, so the form just completed and updated both values to be the same.

Reply:
This is false when the processing dataset is the same as the local storage dataset. It will be false when the page is refreshed.